controller/network/path.go

```diff
@@ -41,13 +41,17 @@ func (self *Path) String() string {
 		return "{malformed}"
 	}
```

Comment:
It's a bit complicated b/c there may be zero links if we're initiating and terminating the connection on a single router.

Reply:
I didn't think of that. Should have probably done it like this in the first place. Added a check if there are any links to conditionally print and added end of slice check to print within the loop.